airflow/models/dagrun.py

```diff
@@ -839,25 +839,43 @@ def task_filter(task: "Operator") -> bool:
 
         if hook_is_noop:
 
-            def create_ti_mapping(task: "Operator") -> dict:
-                created_counts[task.task_type] += 1
-                return TI.insert_mapping(self.run_id, task, map_index=-1)
+            def create_ti_mapping(task: "Operator") -> List[dict]:
+                if isinstance(task, MappedOperator):
```

Comment:
```suggestion
                if task.is_mapped:
```

Reply:
Should I move `is_mapped` to BaseOperator because mypy complains(on the code you gave me too @ashb which is same as the review above)

Reply:
Complains with what error specifically, because it is there https://github.com/apache/airflow/blob/436c17c655494eff5724df98d1a231ffa2142253/airflow/models/baseoperator.py#L1479

A useful thing: what type does mypy think `task` is here.
```suggestion
                reveal_type(task)
                if task.is_mapped:
```

And then run mypy, and it will print out info about `task` at mypy (this is not a real function, nothing is needed to be imported, it is just for mypy)

Reply:
Have pushed the code. Revealed type is "Union[airflow.models.baseoperator.BaseOperator, airflow.models.mappedoperator.MappedOperator]". Quite confusing

Reply:
Have pushed the code. Revealed type is "Union[airflow.models.baseoperator.BaseOperator, airflow.models.mappedoperator.MappedOperator]". Quite confusing

Reply:
I have seen why this is happening. `vars(mapped)` task returns an empty dict.

Reply:
That is because is_mapped is a property on the class (so it's not set in the `__dict__` of the object). What is the exact error mypy gives please?

Reply:
Have type ignored it.
```
airflow/models/dagrun.py:828: error: Item "BaseOperator" of
"Union[BaseOperator, MappedOperator]" has no attribute
"mapped_args_are_literals"  [union-attr]
                if task.is_mapped and task.mapped_args_are_literals:
                                      ^
airflow/models/dagrun.py:830: error: Item "BaseOperator" of
"Union[BaseOperator, MappedOperator]" has no attribute "x"  [union-attr]
                    if ti.map_index >= task.x:
                                       ^
airflow/models/dagrun.py:870: note: Revealed type is "Union[airflow.models.baseoperator.BaseOperator, airflow.models.mappedoperator.MappedOperator]"
airflow/models/dagrun.py:871: error: Item "BaseOperator" of
"Union[BaseOperator, MappedOperator]" has no attribute
"mapped_args_are_literals"  [union-attr]
                if not task.is_mapped or not task.mapped_args_are_literals...
                                             ^
airflow/models/dagrun.py:873: error: Incompatible return value type (got
"Tuple[Union[BaseOperator, MappedOperator], range]", expected
"Tuple[Union[BaseOperator, MappedOperator], Tuple[int, ...]]")  [return-value]
                return (task, range(task.x))
                        ^
airflow/models/dagrun.py:873: error: Item "BaseOperator" of
"Union[BaseOperator, MappedOperator]" has no attribute "x"  [union-attr]
                return (task, range(task.x))
                                    ^
Found 5 errors in 1 file (checked 1 source file)

ERROR: The previous step completed with error. Please take a look at output above 
```